sdk/endpoint/compositeBuilder.go

```diff
@@ -0,0 +1,24 @@
+package endpoint
+
+type CompositeEndpointBuilder interface {
+	Append(...ChainedEndpoint) CompositeEndpointBuilder
+	Build() *CompositeEndpoint
+}
+
+func NewCompositeEndpointBuilder() CompositeEndpointBuilder {
+	return &compositeEndpointBuilder{
+		endpoints: make([]ChainedEndpoint, 0, 4),
+	}
```

Comment:
Interesting, it surely makes buidling composites smoother. I am willing to accept this, but shallwe consider adding more methods to the interface: Insert and Remove fore example. Let's make these complete now and don't change it later.

Reply:
@nickolaev Could you suggest an example of where we'd use Insert or Remove ?